File: FindMissingAndRepeatedValues/findMissingAndRepeatedValues.py
from collections import defaultdict
from typing import List


def findMissingAndRepeatedValues(grid: List[List[int]]) -> List[int]:
    count = defaultdict(int)

    for i in grid:
        for j in i:
            count[j] += 1

    a, b = 0, 0
    for i in range(1, (len(grid) * len(grid) + 1)):
        if count[i] == 2:
            a = i
        elif count[i] == 0:
            b = i
    return [a, b]

File: FindMissingAndRepeatedValues/findMissingAndRepeatedValues.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function(grid) {
    const N = grid.length * grid.length;
    const count = new Array(N + 1).fill(0);
    // for (let i = 0; i < grid.length; i++) {
    //     for (let j = 0; j < grid.length; j++) {
    //         count[grid[i][j]] += 1;
    //     }
    // }
    for (const row of grid) {
        for (const element of row) {
            count[element] += 1;
        }
    }
    let a = 0;
    let b = 0;
    for (let index = 1; index < N + 1; index++) {
        if (count[index] > 1) {
            a = index;
        } else if (count[index] == 0) {
            b = index;
        }

    }
    return [a, b];
};

console.log(findMissingAndRepeatedValues([[9,1,7],[8,9,2],[3,4,6]])); // [6, 9]